packages/glibc227.rb

```diff
@@ -0,0 +1,134 @@
+        system "sed -i 's,char \\*loc2,char \\*loc2 __attribute__ ((nocommon)),g' regexp.c"
+        system "sed -i 's,char \\*locs,char \\*locs __attribute__ ((nocommon)),g' regexp.c"
+      end
+    end
+  end
+
+  def self.build
+    system "mkdir -pv glibc_build"
+    Dir.chdir "glibc_build" do
+      case ARCH
+      when 'armv7l' || 'aarch64'
+        system "../configure",
+               "--prefix=#{CREW_PREFIX}",
+               "--libdir=#{CREW_LIB_PREFIX}",
+               "--with-headers=#{CREW_PREFIX}/include",
+               "--disable-werror",
+               "--disable-sanity-checks",
+               "--enable-shared",
+               "--enable-obsolete-rpc",
+               "libc_cv_forced_unwind=yes"
+      when 'x86_64'
+        system "echo \"slibdir=#{CREW_LIB_PREFIX}\" > configparms"
+        puts "echo \"slibdir=#{CREW_LIB_PREFIX}\" > configparms"
+        system "../configure",
+               "--prefix=#{CREW_PREFIX}",
+               "--libdir=#{CREW_LIB_PREFIX}",
+               "--with-headers=#{CREW_PREFIX}/include",
+               "--disable-werror",
+               "--disable-sanity-checks",
+               "--enable-shared",
+               "--disable-multilib",
+               "--enable-obsolete-rpc",
+               "libc_cv_forced_unwind=yes",
+               "libc_cv_ssp=no",
+               "libc_cv_ssp_strong=no"
+      when 'i686'
+        system "../configure",
+               "--prefix=#{CREW_PREFIX}",
+               "--libdir=#{CREW_LIB_PREFIX}",
+               "--with-headers=#{CREW_PREFIX}/include",
+               "--disable-werror",
+               "--disable-sanity-checks",
+               "--enable-shared",
+               "--disable-multilib",
+               "--enable-obsolete-rpc",
+               "libc_cv_forced_unwind=yes",
+               "libc_cv_ssp=no",
+               "libc_cv_ssp_strong=no"
+      end
+      system "make"
+    end
+  end
+
+  def self.install
+    Dir.chdir "glibc_build" do
+      system "mkdir -pv #{CREW_DEST_PREFIX}/etc"
+      system "touch","#{CREW_DEST_PREFIX}/etc/ld.so.conf"
+      system "make", "DESTDIR=#{CREW_DEST_DIR}", "install"
+      system "make", "DESTDIR=#{CREW_DEST_DIR}", "localedata/install-locales"  # This installs full locales, but it has large disk size.
+    end
+      
+    # minimum set of locales -> /usr/local/lib(64)/locale/locale-archive
+#    Dir.chdir "glibc_build/localedata" do
+#      system "mkdir -pv #{CREW_DEST_LIB_PREFIX}/locale"
+#      puts "Install minimum set of locales".lightblue
+      
+      # Assume old version of glibc is installed. -> use localedef.
+      # If not installed, we can move following instruction to postinstall
+      # Since glibc is a basic package, we prefer to provide pre-compiled package.
+      # No compilcated detect logics required -> make it as simple as possible
+      system "localedef --prefix=#{CREW_DEST_DIR} -i cs_CZ -f UTF-8 cs_CZ.UTF-8"
```

Comment:
I'm thinking this should be `--prefix=#{CREW_DEST_PREFIX}`?

Reply:
Would it be CREW_PREFIX or CREW_DEST_PREFIX?

Reply:
`#{CREW_DEST_PREFIX}` seems the better choice since the files need tracked.

Reply:
They're being installed to the dest directory.
```Never use cat on a binary file``

Reply:
Here:
```.././scripts/mkinstalldirs /usr/local/tmp/crew/dest/usr/local/lib64/locale
mkdir -p -- /usr/local/tmp/crew/dest/usr/local/lib64/locale```